src/openforms/contrib/objects_api/validators.py

```diff
@@ -78,3 +91,43 @@ def validate_document_type_references(
 
     if errors:
         raise ValidationError(errors)
+
+
+def validate_object_ownership(form: Form, session, initial_data_reference: str) -> None:
+    auth_info = session.get(FORM_AUTH_SESSION_KEY)
+    if not auth_info:
+        raise PermissionDenied("Cannot pass data reference as anonymous user")
+
+    object = None
+    for backend in form.registration_backends.filter(backend="objects_api"):
+        if not backend.options:
+            continue
+
+        api_group = ObjectsAPIGroupConfig.objects.filter(
+            pk=backend.options.get("objects_api_group")
+        ).first()
+        if not api_group:
+            continue
+
+        try:
+            with get_objects_client(api_group) as client:
+                try:
+                    object = client.get_object(initial_data_reference)
+                    break
```

Comment:
Why is this needed here?I mean, don't we exit the loop if there is no exception?

Reply:
It's needed because we should exist the loop once an object is found, to not do unnecessary fetches to services from other registration backends.